File: Java/app/src/main/java/software/rsquared/template/di/ActivityBuilderModule.java
package software.rsquared.template.di;


import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import software.rsquared.template.ui.main.MainActivity;

/**
 * @author rSquared.software
 */
@Module
abstract class ActivityBuilderModule {

//	@ContributesAndroidInjector(modules = {AuthFragmentsBuilder.class})
//	abstract AuthActivity bindAuthActivity();

	@ContributesAndroidInjector(modules = {MainFragmentsBuilder.class, MainActivityModule.class})
	abstract MainActivity bindMainActivity();

}

File: Kotlin/app/src/main/java/software/rsquared/template/di/ActivityBuilderModule.kt
package software.rsquared.template.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import software.rsquared.template.ui.main.MainActivity

/**
 * @author Rafal Orlik
 */
@Module
abstract class ActivityBuilderModule {

    @ContributesAndroidInjector(modules = [MainModule::class, MainFragmentsBuilder::class])
    internal abstract fun contributeMainActivity(): MainActivity
}